homeassistant/components/shelly/climate.py

```diff
@@ -308,6 +318,9 @@ async def async_added_to_hass(self) -> None:
         LOGGER.info("Restoring entity %s", self.name)
 
         last_state = await self.async_get_last_state()
+        last_extra_data = await self.async_get_last_extra_data()
+        if last_extra_data is not None:
+            self._last_target_temp = last_extra_data.as_dict()["last_target_temp"]
```

Comment:
Maybe move this to above line 332 so that the `last_state` handling is not breaked in the middle to improve readability?